mlrun/feature_store/feature_set.py

```diff
@@ -391,6 +403,52 @@ def purge_targets(self, target_names: List[str] = None, silent: bool = False):
             self, mlrun.api.schemas.AuthorizationAction.delete
         )
 
+        purge_targets = self._reload_and_get_status_targets(
+            target_names=target_names, silent=silent
+        )
+
+        if purge_targets:
+            # TODO - benb - run_uuid shouldn't be a part of purge - remove logic after toggle implementation
+            from mlrun.datastore.targets import generate_target_run_uuid
+
+            run_uuid = generate_target_run_uuid()
+            purge_target_names = list(purge_targets.keys())
+            for target_name in purge_target_names:
+                target = purge_targets[target_name]
+                driver = get_target_driver(target_spec=target, resource=self)
+                try:
+                    driver.purge()
+                    driver.run_uuid = run_uuid
+                except FileNotFoundError:
+                    pass
+                del self.status.targets[target_name]
+
+            self.save()
+
+    def update_targets_for_ingest(
+        self,
+        targets: List[DataTargetBase],
+        overwrite: bool = None,
+    ):
+        ingestion_target_names = [t if isinstance(t, str) else t.name for t in targets]
+
+        status_targets = {}
+        if not overwrite:
+            # silent=True always because targets are not guaranteed to be found in status
+            status_targets = (
+                self._reload_and_get_status_targets(
+                    target_names=ingestion_target_names, silent=True
+                )
+                or {}
+            )
+
+        from mlrun.datastore.targets import update_targets_for_ingest
```

Comment:
Why are you importing here and not at the file beginning? It's highly irregular - please add a comment explaining why this is needed.
See my next comment, though - I don't understand why this is needed at all.

Reply:
Moved it to the beginning of the method, this is common coding on our project when this import is only used in that block of code on that file. See line 771 for example in the same file.

Reply:
In general, we use imports at the beginning of the file unless there's a specific reason to place them elsewhere (circular dependencies being the main reason). 
Specifically in this case I still don't understand why this import is needed at all, since the method you're importing (`update_targets_for_ingest`) is only called here across the entire code-base. Given it's specific logic for this flow, why not putting it in this method or alternatively put it in this class rather than globally?